### PR TITLE
explicitly inherit "nofile" limit from crio daemon

### LINKED ISSUE #4807: ulimit values are not inherited to runtimeVM
**Description**

According to documentation if no ulimit values were set _"settings will be inherited from the CRI-O daemon"_, however, it seems to be wrong for some runtimes such as kata-runtime, where ulimits are not inherited from CRI-O.
This may also cause application failures if nofile value is to large, [see this issue](https://github.com/kata-containers/runtime/issues/2597).

**Steps to reproduce the issue:**
1.[ install kata](https://github.com/kata-containers/documentation/tree/master/install)
2. run some deployment with kata as runtime and without
3. check for differences in ulimit -a

([alternatively see](https://github.com/kata-containers/runtime/issues/2597))

**Describe the results you received:**
native runtime:
```
$ kubectl exec pod/fedora -- ulimit -n
1048576
```

kata runtime:
```
$ kubectl exec pod/fedora-kata -- ulimit -n
1073741816
```

**Describe the results you expected:**
same value

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `crio --version`:**

```
crio version 1.21.0
Version:       1.21.0
GitCommit:     40d6ec0f56215417267f5297c99bdab3cc36827c
GitTreeState:  clean
BuildDate:     2021-04-26T13:51:41Z
GoVersion:     go1.14.15
Compiler:      gc
Platform:      linux/amd64
Linkmode:      dynamic
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

### PR DESCRIPTION
**Pull request by snir911**
large "nofile" value may cause application running issues, lets
explicitly inherit it, as it's not natively inherited in all runtimes
as documented.

Signed-off-by: Snir Sheriber <ssheribe@redhat.com>

#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
Fixes #4807 
#### Which issue(s) this PR fixes:

Fixes #4807 

[also related](https://github.com/kata-containers/runtime/issues/2597)

#### Special notes for your reviewer:
`None`
#### Does this PR introduce a user-facing change?
`None`

```release-note
`None`
```

### CODE REVIEW

**Comment by fidencio** [comment]
@haircommander, @saschagrunert, @kolyshkin, would you mind to take a look at this one?

**Comment by haircommander** [comment]
see my comment https://github.com/cri-o/cri-o/issues/4807#issuecomment-826902914
I am not sure yet, but I may be a NACK on this one

File: docs/crio.conf.5.md [hunk 1]
```diff
@@ -99,7 +99,7 @@ The `crio.runtime` table contains settings pertaining to the OCI runtime used an
   The _name_ of the OCI runtime to be used as the default.
 
 **default_ulimits**=[]
-  A list of ulimits to be set in containers by default, specified as "<ulimit name>=<soft limit>:<hard limit>", for example:"nofile=1024:2048". If nothing is set here, settings will be inherited from the CRI-O daemon.
+  A list of ulimits to be set in containers by default, specified as "<ulimit name>=<soft limit>:<hard limit>", for example:"nofile=1024:2048". If nothing is set here, limits that aren't explicitly set by runtime-tools will be inherited from the process launching the container.
```
Comment (TomSweeneyRedHat) [inline docs/crio.conf.5.md:102]: It's not wrong, but I've found it better to limit contractions, especially ones that are less frequently used, in man pages.  They are perfectly correct but less understandable for those who did not grow up speaking English.  My $.02.
```suggestion
  A list of ulimits to be set in containers by default, specified as "<ulimit name>=<soft limit>:<hard limit>", for example:"nofile=1024:2048". If nothing is set here, limits that are not explicitly set by runtime-tools will be inherited from the process launching the container.
```

File: pkg/config/template.go [hunk 2]
```diff
@@ -652,7 +652,7 @@ const templateStringCrioRuntime = `# The crio.runtime table contains settings pe
 const templateStringCrioRuntimeDefaultUlimits = `# A list of ulimits to be set in containers by default, specified as
 # "<ulimit name>=<soft limit>:<hard limit>", for example:
 # "nofile=1024:2048"
-# If nothing is set here, settings will be inherited from the CRI-O daemon
+# If nothing is set here, limits that aren't explicitly set by runtime-tools will be inherited from the process launching the container.
```
Comment (TomSweeneyRedHat) [inline pkg/config/template.go:655]: If you change the above, please change it here too.

**Pull request closed without merging**